cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -322,6 +325,9 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
           twillPreparer.addJVMOptions(jvmOpts);
         }
 
+        // Overwrite JVM options if specified
+        LauncherUtils.overrideJVMOpts(cConf, twillPreparer, runnables);
```

Comment:
Can't we call the override JVM options in KubeTwillPreparer for all runnables, instead of calling it in every runnable? What is the downside of that?

Reply:
We can, but then the logic for overriding JVM options will only work for KubeTwillPreparer and not other TwillPreparer implementations.

Reply:
This could be something that would benefit from an abstract class potentially; it would depend on what componet we wanted to be responsible for this behavior.

Reply:
Makes sense. thanks.